src/Visitor.php

```diff
@@ -0,0 +1,28 @@
+<?php
+
+namespace HexletPsrLinter;
+
+use PhpParser\Node;
+use PhpParser\NodeVisitorAbstract;
+
+class Visitor extends NodeVisitorAbstract
+{
+    private $errorFunctions = [];
+
+    public function leaveNode(Node $node)
+    {
+        $nodeClass = get_class($node);
+        if ($nodeClass == "PhpParser\\Node\\Stmt\\ClassMethod" && checkFunctionName($node->name) != true) {
```

Comment:
Поделитесь пожалуйста ссылкой на эти статические свойства. А то я не могу
загуглить. :: гугл не понимает.

On Fri, Aug 5, 2016 at 10:19 AM, Roman Lakhtadyr <notifications@github.com>
wrote:

> In src/Visitor.php
> <https://github.com/maximtop/hexlet-psr-linter/pull/1#discussion_r73651466>
> :
>
> > @@ -0,0 +1,28 @@
> > +<?php
> > +
> > +namespace HexletPsrLinter;
> > +
> > +use PhpParser\Node;
> > +use PhpParser\NodeVisitorAbstract;
> > +
> > +class Visitor extends NodeVisitorAbstract
> > +{
> > +    private $errorFunctions = [];
> > +
> > +    public function leaveNode(Node $node)
> > +    {
> > +        $nodeClass = get_class($node);
> > +        if ($nodeClass == "PhpParser\\Node\\Stmt\\ClassMethod" && checkFunctionName($node->name) != true) {
>
> Начиная с PHP 5.6 у каждого класса есть статическое свойство ::class
> которое содержит в себе имя класса. Поэтому лучше использовать именно это
> свойство а не записывать имя в виде строки вручную
> "PhpParser\\Node\\Stmt\\ClassMethod".
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/maximtop/hexlet-psr-linter/pull/1/files/de77fd2c026aec74dad587f6eb61c5e151b5f7c4#r73651466>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AQePwUOmAr6bruMqbOyC3y-tpTn99Nqgks5qcuPogaJpZM4JdWB2>
> .
>



-- 
Sincerely yours,
Maxim Nikolaevich
mob. 8-926-934-08-25
email: maximtop@gmail.com